src/hb-ot-layout-common.hh

```diff
@@ -343,6 +475,44 @@ struct LangSys
     return_trace (c->embed (*this));
   }
 
+  bool operator == (const LangSys& o) const
+  {
+    if (featureIndex.len != o.featureIndex.len ||
+        reqFeatureIndex != o.reqFeatureIndex)
+      return false;
+
+    bool has_diff = false;
+    for (const auto& _ : + hb_zip (featureIndex, o.featureIndex))
+    {
+      if (_.first != _.second)
+      {
+        has_diff = true;
+        break;
+      }
+    }
+    return has_diff == false;
+  }
+
+  bool subset (hb_subset_context_t *c,
+	       const hb_map_t      *feature_index_map) const
+  {
+    TRACE_SUBSET (this);
+    auto *out = c->serializer->start_embed (*this);
+    if (unlikely (!out || !c->serializer->extend_min (out))) return_trace (false);
+
+    out->reqFeatureIndex = hb_map_has (feature_index_map, reqFeatureIndex) ? hb_map_get (feature_index_map, reqFeatureIndex) : 0xFFFFu;
```

Comment:
You can use the has() and get() methods directly on the map object (ie. feature_index_map->has(reqFeatureIndex))